### PR TITLE
🔍 feat: Filter MultiSelect and SelectDropDown (+variants) + CSS fixes for Scrollbar

### PR DESCRIPTION
**Pull request by danny-avila**
Originally https://github.com/danny-avila/LibreChat/pull/2090

Took opportunity to debounce inputs from conversation settings (just for OpenAI for now, which also covers azure and custom endpoints)

Resolve footer React key map error

refactored mouse/enter with useCallback

memoize conversations to prevent re-render when props don't change

resolve other misc. react errors related to unrecognized props

### CODE REVIEW

**Comment by danny-avila** [comment]
> The `React.Fragment` ones in particular feel less necessary with modern React, so I wanted to understand the choice to add keys to those elements a bit better. _Usually,_ manually keying static renders like that is useful in volatile ordering situations (something I've had to do at work for various WYSIWYG implementations, as well as our page editing software) - but was curious if you saw something going on in react dev tools that made it seem like keys were needed here.

Without using React.Fragment, there is a big error block in the console for not using keys with mapped elements. It may not make a meaningful rendering difference, but it does resolve this React error.

**Comment by danny-avila** [comment]
@f1yn FYI, the MultiSearch seems to ignore spacebar input within the Preset/Conversation Settings. 

I didn't want to spend too much time on this but an easy fix was not apparent.

it's ok as most if not all model names don't have spaces, and I think filtering by first few letters is good enough in this case.

File: client/src/style.css [hunk 1]
```diff
@@ -1107,35 +1094,32 @@ button {
...
 }
 
-.scrollbar-transparent::-webkit-scrollbar-thumb {
-  background-color: rgba(0, 0, 0, 0.1);
+::-webkit-scrollbar-track {
+  background-color: transparent;
+  border-radius: 9999px;
 }
 
-@media (prefers-color-scheme: dark) {
```
Comment (f1yn) [inline client/src/style.css]: Hey, was there a particular reason this was removed? This was the original fix for the issue described with lightmode, as it's dependent on the code preference of the window.

Comment (f1yn) [reply client/src/style.css]: Does `.dark` negate the need for this?

Comment (danny-avila) [reply client/src/style.css]: Yes, your fix wasn’t working for me as expected so I changed the approach here

**Comment by fuegovic** [comment]
Two small things I noticed:

The search box, when used in the plugin list still use "Search model by name"
image

And the dashes `-` aren't filtered out of the search (On the demo, in the OpenRouter list, I use dashes as separators between different categories and this is how search comes out)
image

**Comment by fuegovic** [comment]
> @fuegovic can you DM me tonight with the details (specifically the demo details)? I can likely take a look tonight.

Thanks @f1yn 
The demo is public, you can create an account on it if you want to check it out yourself: https://demo.librechat.cfd/
you can DM me on discord anytime if you need more details

**Comment by f1yn** [comment]
Just an update, I'm not going to have time tonight to take a look, but I should have some time tomorrow.

@fuegovic I think we can adjust the new hook's key function avoid to avoid matching any keys that return null.

My current plan involves [updating the filter](https://github.com/danny-avila/LibreChat/pull/2138/files#diff-c6580f25a7b68313b416559a63a6fed91e3978be8468a2872f31e5e8747ddd26R91) function to detect a null return, and then return `false` for that specific list of elements.

As for the whitespace, that's either caused by a problem with the input, or bad event handling on an ancesor component. I can also take a look at that, but I imagine the culprit would be in either of those two places.

I appreciate your patience with this, and I'm sorry I won't be able to promptly look into it. I will let you know when I get to itt tomorrow.

**Comment by fuegovic** [comment]
@f1yn no worries, thanks for looking into this!

**Comment by danny-avila** [comment]
> Just an update, I'm not going to have time tonight to take a look, but I should have some time tomorrow.
> 
> @fuegovic I think we can adjust the new hook's key function avoid to avoid matching any keys that return null. In this case, we'd detect items that have more than (4) consecutive `-`
> 
> My current plan involves [updating the filter](https://github.com/danny-avila/LibreChat/pull/2138/files#diff-c6580f25a7b68313b416559a63a6fed91e3978be8468a2872f31e5e8747ddd26R91) function to detect a null return, and then return `false` for that specific list of elements.
> 
> As for the whitespace, that's either caused by a problem with the input, or bad event handling on an ancesor component. I can also take a look at that, but I imagine the culprit would be in either of those two places.
> 
> I appreciate your patience with this, and I'm sorry I won't be able to promptly look into it. I will let you know when I get to it tomorrow.

Thank you for looking into this, I appreciate it!

**Pull request merged**